src/ElasticSearchRepository.php

```diff
@@ -44,34 +47,42 @@ public function __construct(
         Client $client,
         Serializer $serializer,
         string $index,
-        string $class,
+        string $class_type,
         array $notAnalyzedFields = []
     ) {
         $this->client = $client;
         $this->serializer = $serializer;
         $this->index = $index;
-        $this->class = $class;
+        $this->class_type = $class_type;
         $this->notAnalyzedFields = $notAnalyzedFields;
     }
 
     /**
-     * {@inheritdoc}
+     * @param Identifiable $data
+     * @throws AssertionFailedException
      */
     public function save(Identifiable $data): void
     {
-        Assertion::isInstanceOf($data, $this->class);
+        if (!class_exists($this->class_type)) {
```

Comment:
If we want to check if this class exists, we should do it in the constructor.

Reply:
:) My thoughts exactly, but then phpstan is not aware of the check and raise the exception again